src/db/sysdb_search.c

```diff
@@ -666,7 +666,9 @@ errno_t sysdb_search_with_ts_attr(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    if (optflags & SYSDB_SEARCH_WITH_TS_ONLY_SYSDB_FILTER) {
+    switch (search_cache) {
+    case SYSDB_CACHE_TYPE_PERSISTENT:
+    {
```

Comment:
Are the blocks in the cases needed? They look visually distracting to me and the switch blocks and the case blocks have the same indentation level..

Reply:
There are two variables declaration in those cases. switch() cases are not considered to be scope blocks itself and to be able to declare something there explicit scope is needed. When I move those declarations out of switch() statement they looks to be detached and out of purpose. Mentioned declarations are in line691 and line 744.

Reply:
Umm OK, fair enough, do you think it would be better to split the cases into their own functions rather?

Reply:
I think that if we do that sysdb_search_with_ts_attr() function will be only sort of thin proxy to those two. I think this is topic for more general code refactoring to make those whole routines more "flat".